### PR TITLE
Stop embedding iOS framework

### PR DESCRIPTION
**Pull request by felipe-saez**
This PR includes the following changes:

- Removes embedded iOS XCFramework
- Creates Pre Processor script to temporarily download XCFramework into main project
- Updates Post Processor script to import XCFramework into Xcode project
- Uses `iosNativeVersion` property from `package.json` file to avoid hardcoding version of iOS XCFramework

### CODE REVIEW

File: source/Assets/Plugins/Didomi/Editor/PostProcessor.cs [hunk 1]
```diff
@@ -191,6 +194,7 @@ public static void OnPostProcessBuild(BuildTarget buildTarget, string buildPath)
         
         if (buildTarget == BuildTarget.iOS)
         {
+
```
Comment (pmerlet-at-didomi) [inline source/Assets/Plugins/Didomi/Editor/PostProcessor.cs:197]: Additional line

File: source/Assets/Plugins/Didomi/Editor/PostProcessor.cs [hunk 2]
```diff
@@ -212,14 +216,88 @@ public static void OnPostProcessBuild(BuildTarget buildTarget, string buildPath)
...
+            var unityTargetGuid = proj.GetUnityFrameworkTargetGuid();
+            string fileGuid = proj.AddFile(frameworkPath, "Frameworks/" + Path.GetFileName(frameworkPath), PBXSourceTree.Source);
+            proj.AddFileToBuild(unityTargetGuid, fileGuid);
+
             SetupDidomiFrameworkForTargetSDK(proj, targetGuid, buildPath);
             CopyDidomiConfigFileToIOSFolder(proj, targetGuid, buildPath);
             CopyPackageJsonToIOSFolder(proj, targetGuid, buildPath);
 
             proj.WriteToFile(projPath);
+
```
Comment (pmerlet-at-didomi) [inline source/Assets/Plugins/Didomi/Editor/PostProcessor.cs:237]: Additional lines

File: source/Assets/Plugins/Didomi/Editor/PostProcessor.cs [hunk 3]
```diff
@@ -375,5 +453,4 @@ public static void ReplaceLineInFile(string path, string oldValue, string newVal
         text = text.Replace(oldValue, newValue);
         File.WriteAllText(path, text);
     }
-}
-
+}
```
Comment (pmerlet-at-didomi) [inline source/Assets/Plugins/Didomi/Editor/PostProcessor.cs:456]: Missing line break

File: source/Assets/Plugins/Didomi/Editor/DidomiPaths.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
+using System.IO;
+using UnityEngine;
+
+/// <summary>
+/// Properties and methods related to paths used to handle Didomi dependencies.
+/// </summary>
+public static class DidomiPaths
```
Comment (pmerlet-at-didomi) [inline source/Assets/Plugins/Didomi/Editor/DidomiPaths.cs:7]: The file name should probably be changed, as it is not only related to paths, and is specific to iOS currently. Maybe something like `DidomiIOSDependencies`?

Comment (felipe-saez) [reply source/Assets/Plugins/Didomi/Editor/DidomiPaths.cs:7]: I'll be pushing an update that will change the references to this class from things like `DidomiPaths.FRAMEWORK_URL` to `DidomiPaths.IOS.FRAMEWORK_URL`. That way we can keep properties that are common to all platforms still under DidomiPaths, for example `DidomiPaths.PACKAGE_JSON_PATH` and the Android specific Paths under `DidomiPaths.Android`.

File: source/Assets/Plugins/Didomi/Editor/XCFrameworkUtils.cs [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        }
+        
+        // Create an unzip command string to be executed by bash shell using the 'unzip' command, which extracts files from a ZIP archive 
+        string unzipFramework = $"echo \"unzipping\" && unzip {DidomiPaths.IOS.ZIP_FILE_PATH} -d {DidomiPaths.IOS.FRAMEWORK_DIRECTORY_IN_SOURCE}";
+        BashUtils.Execute(unzipFramework);
+
+        // Delete zip file since it's not needed after the framework has been unzipped.
+        FileUtils.DeleteImportedFile(DidomiPaths.IOS.ZIP_FILE_PATH);
+    }
+
```
Comment (nicolas-chaix-didomi) [inline source/Assets/Plugins/Didomi/Editor/XCFrameworkUtils.cs:47]: duplicate empty line

**Review by nicolas-chaix-didomi — APPROVED** [review]

**Pull request merged**